src/foundry/client-esm/canvas/sources/rendered-effect-source.d.mts

```diff
@@ -0,0 +1,325 @@
+import type { InexactPartial } from "../../../../types/utils.d.mts";
+import BaseEffectSource from "./base-effect-source.mjs";
+
+// TODO: Remove after shaders are done
+type AdaptiveDarknessShader = unknown;
+
+/**
+ * An abstract class which extends the base PointSource to provide common functionality for rendering.
+ * This class is extended by both the LightSource and VisionSource subclasses.
+ */
+declare class RenderedEffectSource<
+  SourceData extends RenderedEffectSource.RenderedEffectSourceData = RenderedEffectSource.RenderedEffectSourceData,
+  SourceShape extends PIXI.Polygon = PIXI.Polygon,
+  RenderingLayers extends Record<string, RenderedEffectSource.RenderedEffectSourceLayer> = RenderedEffectSource.Layers,
+> extends BaseEffectSource<SourceData, SourceShape> {
+  /**
+   * Keys of the data object which require shaders to be re-initialized.
+   */
+  protected static _initializeShaderKeys: string[];
+
+  /**
+   * Keys of the data object which require uniforms to be refreshed.
+   */
+  protected static _refreshUniformsKeys: string[];
+
+  /**
+   * Layers handled by this rendered source.
+   */
+  protected static get _layers(): Record<string, RenderedEffectSource.RenderedEffectLayerConfig>;
```

Comment:
I assume this is equal to `RenderingLayers`? Since this is on the static side, yeah we can't guarantee that statically.

Consider adding a remark about that.

Reply:
Not *quite*, the values are different but the keys are supposed to match